macro/src/class_info.rs

```diff
@@ -238,6 +361,31 @@ pub enum NonRepeatingType {
     Scalar(ScalarType),
 }
 
+impl NonRepeatingType {
+    pub fn descriptor(&self) -> String {
+        match self {
+            NonRepeatingType::Ref(r) => match r {
+                RefType::Class(c) => format!("L{};", c.name.with_slashes()),
+                RefType::Array(r) => format!("[{}", r.descriptor()),
+                RefType::TypeParameter(_)
+                | RefType::Extends(_)
```

Comment:
I'm assuming the full Java generics story won't be solved here, but one note that the `<T extends SomeClass>` won't translate to a `Ljava/lang/Object;` in a method signature. For example

```java
public <E extends java.lang.Number> void add(E number) { }
```

compiles as 

```
public <E extends java.lang.Number> void add(E);
    descriptor: (Ljava/lang/Number;)V 
```

and I think looking up the method by signature in the JNI will fail if we used `java.lang.Object`.

Reply:
Oh, hmm, good point. That's annoying. What happens if you do `extends java.lang.Number & SomeInterface`, for example?

Reply:
I had to go write a test case to confirm because this one surprised me... `extends java.lang.Number & SomeInterface` is just `Ljava.lang.Number;`. For `extends T1 & T2 & ... TN`, it's compiled as `LT1;` regardless of any relationship between the types in the extends set.

Reply:
Tracking in https://github.com/duchess-rs/duchess/issues/42